context.go

```diff
@@ -1,16 +1,49 @@
 package basictracer
 
-// Context holds the basic Span metadata.
-type Context struct {
+import (
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+// SpanContext holds the basic Span metadata.
+type SpanContext struct {
 	// A probabilistically unique identifier for a [multi-span] trace.
 	TraceID uint64
 
 	// A probabilistically unique identifier for a span.
 	SpanID uint64
 
-	// The SpanID of this Context's parent, or 0 if there is no parent.
-	ParentSpanID uint64
```

Comment:
It's important to note that the parent can be removed from the SpanContext here and moved into the RawSpan *without* ugly / confusing extra parameters to `Join`... this is the big difference with the first go-around from many months ago (where `Join` and `Inject` were more symmetrical).